tests/qos/test_qos_sai.py

```diff
@@ -136,6 +139,133 @@ def testQosSaiPfcXoffLimit(
+        pfcwd_test_port_id = dutConfig["testPorts"]["src_port_id"]
+        pfcwd_test_port = dutConfig["dutInterfaces"][pfcwd_test_port_id]
+        fanout_neighbors = conn_graph_facts["device_conn"][duthost.hostname]
+        peerdevice = fanout_neighbors[pfcwd_test_port]["peerdevice"]
+        peerport = fanout_neighbors[pfcwd_test_port]["peerport"]
+        peer_info = {
+            'peerdevice': peerdevice,
+            'hwsku': fanout_graph_facts[peerdevice]["device_info"]["HwSku"],
+            'pfc_fanout_interface': peerport
+        }
+
+        queue_index = qosConfig[xonProfile]["pg"]
+        frames_number = 100000000
+
+        logging.info("PFC Storm Gen Params \n DUT iface: {} Fanout iface : {}\
+                      queue_index: {} peer_info: {}".format(pfcwd_test_port,
+                                                            peerport,
+                                                            queue_index,
+                                                            peer_info))
+
+        # initialize PFC Storm Handler
+        storm_hndle = PFCStorm(duthost, fanout_graph_facts, fanouthosts,
+                               pfc_queue_idx = queue_index,
+                               pfc_frames_number = frames_number,
+                               peer_info = peer_info)
+        storm_hndle.deploy_pfc_gen()
+
+        # stop pfcwd
+        logger.info("--- Stopping Pfcwd ---")
+        duthost.command("pfcwd stop")
+
+        # set poll interval for pfcwd
+        duthost.command("pfcwd interval {}".format(pfcwd_timers['pfc_wd_poll_time']))
+
+        logger.info("--- Start Pfcwd on port {}".format(pfcwd_test_port))
+        start_wd_on_ports(duthost,
+                          pfcwd_test_port,
+                          pfcwd_timers['pfc_wd_restore_time'],
+                          pfcwd_timers['pfc_wd_detect_time'])
+
+        try:
+            logger.info("---  Fill the ingress buffers ---")
+            self.runPtfTest(
+                ptfhost, testCase="sai_qos_tests.PtfFillBuffer", testParams=testParams
+            )
+
+            # Trigger PfcWd
+            storm_hndle.start_storm()
+            logger.info("PfcWd Status: {}".format(duthost.command("pfcwd show stats")["stdout_lines"]))
+            time.sleep(10)
+            storm_hndle.stop_storm()
+            logger.info("PfcWd Status: {}".format(duthost.command("pfcwd show stats")["stdout_lines"]))
+
+            logger.info("---  Enable dst iface and verify if the PFC frames are not sent from src port ---")
+            self.runPtfTest(
+                ptfhost, testCase="sai_qos_tests.PtfReleaseBuffer", testParams=testParams
+            )
+        except Exception as e:
+            raise e
+        finally:
+            self.runPtfTest(
+                ptfhost, testCase="sai_qos_tests.PtfEnableDstPorts", testParams=testParams
+            )
+            logger.info("--- Stopping Pfcwd ---")
+            duthost.command("pfcwd stop")
```

Comment:
please get the pfcwd state and pfc timers configured on the dut prior to the changes and restore back to those original values at the end of the test so as not to affect the subsequent test runs